Engine/ChildItemVisibility.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.ComponentModel;
+using System.Xml.Serialization;
+
+namespace OpenTap
+{
+    /// <summary>  This implements the concept of collapsing or expanding a group of items visually. For example a test step's child step list.</summary>
+    public static class ChildItemVisibility
+    {
+        // The folding property can have 3 states, null, collapsed and expanded.
+        // in order to produce nice XML code, we save the value as a string, but
+        // from the API perspective it looks like a bool?.
+        // It is created as an 'attached property' so that a new property can be introduced
+        // Without creating a gap between TestStep capabilities and ITestStep.
+        
+        
+        internal static readonly DynamicMember VisibilityProperty = new DynamicMember
+        {
+            Name = "ui.fold",
+            Attributes = new object[] { new BrowsableAttribute(false), new DefaultValueAttribute(null), new XmlAttributeAttribute() },
+            DefaultValue = null,
+            TypeDescriptor = TypeData.FromType(typeof(string)),
+            DeclaringType = TypeData.FromType(typeof(ChildItemVisibility)),
+            Writable = true,
+            Readable = true
+        };
+
+        /// <summary>  Set visibility to collapsed or expanded. </summary>
+        /// <param name="obj">The object for which to set visibility.</param>
+        /// <param name="expanded">True: expand, False: collapse, null: revert to default.</param>
+        public static void SetVisibility(object obj, bool? expanded)
+        {
+            VisibilityProperty.SetValue(obj, expanded switch
+            {
+                null => null,
+                true => "expand",
+                false => "collapse"
+            });
+        }
+        /// <summary>  Gets the current visibility state. If not set it will return null. </summary>
+        /// <param name="obj">The object from which to get visibility.</param>
+        public static bool? GetVisibility(object obj)
```

Comment:
Should we make an enum for this instead of a bool?. Similar to WPF vilibility. That should also make it show up nicer in xml without manually doing string stuff.